src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -480,35 +503,82 @@ Transactor::apply()
 NotTEC
 Transactor::checkSign(PreclaimContext const& ctx)
 {
+    // do not check signature of inner batch txn
+    if (ctx.tx.isFieldPresent(sfBatchTxn))
```

Comment:
Is there a test to ensure that including a `BatchTxn` field in a non-Batch-related transaction doesn't succeed? I don't see any code that checks that, but I may not have gotten to that yet.

Reply:
The test is here: https://github.com/XRPLF/rippled/blob/1367e03543171c47af68bc93e370f99e256bcb41/src/test/app/Batch_test.cpp#L1037

The functionality is here: https://github.com/XRPLF/rippled/pull/5060/files#diff-257352f66a3d421d15d2248885c9059bddd6d5f2c7356d95bb91992fcd843eacR1220